src/services/BaseApi.ts

```diff
@@ -0,0 +1,128 @@
+import axios, { AxiosInstance, AxiosRequestConfig, AxiosRequestHeaders, AxiosResponse } from 'axios'
+
+export type PrefectConfig = {
+  baseUrl: string,
+  token?: string,
+}
+
+export type GetApiBaseUrl<T extends PrefectConfig = PrefectConfig> = (config: T) => string
+export type GetApiHeaders<T extends PrefectConfig = PrefectConfig> = (config: T) => AxiosRequestHeaders
+
+export const getPrefectBaseUrl: GetApiBaseUrl = (config) => {
+  return config.baseUrl
+}
+
+export const getPrefectUIHeaders: GetApiHeaders = () => {
+  return {
+    'X-PREFECT-UI': true,
+  }
+}
+
+export const getAuthorizationHeaders: GetApiHeaders = (config) => {
+  const value: AxiosRequestHeaders = {}
+
+  if (config.token) {
+    value.Authorization = `bearer ${config.token}`
+  }
+
+  return value
+}
+
+export class BaseApi<T extends PrefectConfig = PrefectConfig> {
```

Comment:
Since this is not an abstract class I think we can drop the `Base` prefix. Not a rule but I usually expect `BaseThing` classes to not be functional on their own which I believe this class is.